File: Java/Examples/JSONDemo/app/src/main/java/com/example/jsondemo/MainActivity.java
package com.example.jsondemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

/** @Autor Created by Anubys on the 22.03.2020 */


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        DownloadTask task = new DownloadTask(getResources());
        task.execute("https://samples.openweathermap.org/data/2.5/weather?q=London,uk&appid=b1b15e88fa797225412429c1c50c122a1");
    }
}

File: Kotlin/Examples/ViewModelDemo/app/src/main/java/com/anubys/example/viewmodeldemo/MainActivity.kt
package com.anubys.example.viewmodeldemo

/** @Author Created by Anubys on the 20.08.2020 */


import android.os.Bundle
import android.util.Log

import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.ViewModelProvider

import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity(), LifecycleOwner {
    private val tag = MainActivity::class.java.simpleName

    private lateinit var viewModel: MainActivityViewModel
    private lateinit var viewModelFactory: MainActivityViewModelFactory

    //* ************************************************ *
    //*               L I F E - C Y C L E                *
    //* ************************************************ *
    override fun onCreate(savedInstanceState: Bundle?) {
        Log.d(tag,"TAG - MainActivity - onCreate()")
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewModelFactory = MainActivityViewModelFactory("Das ist ein Demo der MainActivityViewModelFactory!")
        viewModel = ViewModelProvider(this, viewModelFactory).get(MainActivityViewModel::class.java)

        tv_display.text = viewModel.number.toString()

        btn_add.setOnClickListener {
            viewModel.addNumber()
            tv_display.text = viewModel.number.toString()
        }
    }
}
